bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSHandler.java

```diff
@@ -328,4 +340,29 @@ private void refreshChannelSubscription(ChannelUID channelUID) {
     public Collection<Class<? extends ThingHandlerService>> getServices() {
         return Collections.singleton(LGWebOSActions.class);
     }
+
+    /**
+     * Make a best effort to automatically detect the MAC address of the TV.
+     * If this does not work automatically, users can still set it manually in the Thing config.
+     */
+    private void findMacAddress() {
+        LGWebOSConfiguration c = getLGWebOSConfig();
+        String host = c.getHost();
+        if (!host.isEmpty()) {
+            try {
+                // validate host, so that no command can be injected
+                String macAddress = WakeOnLanUtility.getMACAddress(InetAddress.getByName(host).getHostAddress());
+                logger.debug("Determined MAC address: {} for host: {}", macAddress, host);
+                if (macAddress != null && !macAddress.equals(c.macAddress)) {
+                    c.macAddress = macAddress;
+                    // persist the configuration change
+                    Configuration configuration = editConfiguration();
+                    configuration.put(LGWebOSBindingConstants.CONFIG_MAC_ADDRESS, macAddress);
+                    updateConfiguration(configuration);
```

Comment:
I was playing with simiilar code in my own PR to update the discovered UDN and this leads to weird behaviour, probably because my thing is defined in a config file (and so with a readonly configuration). I have the feeling that at a certain time (after binding restart), the framework detects a change and restore the original configuration.
But leave it like that until this is more clear for me.

Reply:
OK, let me know if there is a better solution.

Reply:
maybe users with thing file should also put the macAddress into their configuration. So we could update the readme accordingly.